src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -156,6 +156,97 @@ public abstract partial class JsonNamingPolicy
         public static System.Text.Json.JsonNamingPolicy CamelCase { get { throw null; } }
         public abstract string ConvertName(string name);
     }
+    public abstract partial class JsonNode
+    {
+        internal JsonNode() { }
+    }
+    public partial class JsonNumber : System.Text.Json.JsonNode, System.IEquatable<System.Text.Json.JsonNumber>
```

Comment:
Similar to the other JsonNode types, seal JsonNumber as well: https://github.com/dotnet/corefx/pull/40107#discussion_r311718941